app/code/Magento/Customer/view/frontend/templates/form/register.phtml

```diff
@@ -11,6 +11,11 @@ use Magento\Customer\Helper\Address;
 /** @var \Magento\Customer\Block\Form\Register $block */
 /** @var \Magento\Framework\Escaper $escaper */
 $formData = $block->getFormData();
+$showAddressFields = false;
+if ($block->getShowAddressFields() || $block->getConfig('customer/address/show_address_fields')) {
```

Comment:
It would be better to adjust `getShowAddressFields` block method to use newly introduced setting in case if someone created a plugin for it.